packages/@tinacms/graphql/src/build.ts

```diff
@@ -224,3 +234,36 @@ const _buildSchema = async (builder: Builder, tinaSchema: TinaSchema) => {
     ),
   }
 }
+
+function collectionHasReferenceFields(collection: Collection<true>) {
+  let result = false
+  if (!collection.fields) {
+    return false
+  }
+  for (const field of collection.fields) {
+    if (field.type === 'reference') {
+      result = true
```

Comment:
**suggestion**: As `result` cannot be returned to false after line 243, to short-circuit unnecessary loops I suggest:
- removing `result` declaration at line 239.
- returning `true` at this line (line 245)
- returning `false` at line 252.